src/boundary.hpp

```diff
@@ -129,8 +135,8 @@ template<bool O3D> HOST_DEVICE inline void GetBdrySeg(
         bds.xmid = (bds.x + bdinfotb->bd[(bds.Iseg.x + 1) * ny + (bds.Iseg.y + 1)].x)
             * RL(0.5);
 
-        // GlobalLog("Iseg%s %d %d\n", isTop ? "Top" : "Bot", bds.Iseg.x+1, bds.Iseg.y+1);
-        // GlobalLog("Bdryx %g,%g,%g x %g,%g,%g\n", bds.x.x, bds.x.y, bds.x.z, x.x, x.y,
+        // printf("Iseg%s %d %d\n", isTop ? "Top" : "Bot", bds.Iseg.x+1, bds.Iseg.y+1);
+        // printf("Bdryx %g,%g,%g x %g,%g,%g\n", bds.x.x, bds.x.y, bds.x.z, x.x, x.y,
```

Comment:
Just a style question - there are lots of commented out printf's. Are they intended as debugging hints? Otherwise, they could be cleaned up or marked somehow, but not a big deal.

Reply:
When we discover a test which produces different results in the C++/CUDA version compared to the Fortran version, the first step is to isolate this to a single ray, perhaps with `find_failing_ray.py`. After that, both the C++/CUDA and Fortran versions are recompiled with `STEP_DEBUGGING` enabled, which prints the state of the ray after each step and information about why the boundary handling will go the way it does. Once more information has been identified about the cause of the issue, we need to be able to print out intermediate values within the computation. That is what the commented out `printf`s are. They are not all guarded by `STEP_DEBUGGING` because turning them all on would be way too much, but also there's no reason to have to write these `printf` statements from scratch every time we want to look at a particular step.